nntrainer/cl_context.cpp

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Debadri Samaddar <s.debadri@samsung.com>
+ *
+ * @file    cl_context.h
+ * @date    23 Feb 2024
+ * @see     https://github.com/nnstreamer/nntrainer
+ * @author  Debadri Samaddar <s.debadri@samsung.com>
+ * @bug     No known bugs except for NYI items
+ * @brief   This file contains app context related functions and classes that
+ * manages the global configuration of the current OpenCL environment. It also
+ * creates the OpenCL command queue and context.
+ */
+
+#include <cl_context.h>
+#include <fc_layer.h>
+
+namespace nntrainer {
+
+std::mutex cl_factory_mutex;
+
+std::once_flag global_cl_context_init_flag;
+
+static void add_default_object(ClContext &cc) {
+  using LayerType = ml::train::LayerType;
+
+  cc.registerFactory(nntrainer::createLayer<FullyConnectedLayer>,
+                     FullyConnectedLayer::type, LayerType::LAYER_FC);
```

Comment:
seems `LayerType` is only used below. could be removed.

```suggestion
  cc.registerFactory(nntrainer::createLayer<FullyConnectedLayer>,
                     FullyConnectedLayer::type, ml::train::LayerType::LAYER_FC);
```

Reply:
Modified in latest commit